File: 2025-09-September-GFG-POTD/September 30 - Generate all binary strings/Solution.py
class Solution:
    def binstr(self, n: int):
        result = []
        total = 1 << n  # 2^n
        for i in range(total):
            binary = bin(i)[2:]  # remove '0b' prefix
            binary = binary.zfill(n)  # pad with leading zeros
            result.append(binary)
        return result


# Example usage
sol = Solution()
print(sol.binstr(3))  # ['000', '001', '010', '011', '100', '101', '110', '111']

File: 2025-09-September-GFG-POTD/September 30 - Generate all binary strings/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<string> binstr(int n) {
        vector<string> result;
        int total = 1 << n; // 2^n
        for (int i = 0; i < total; i++) {
            string binary = "";
            int num = i;
            // Convert number to binary (padded)
            for (int j = n - 1; j >= 0; j--) {
                binary += ((num >> j) & 1) ? '1' : '0';
            }
            result.push_back(binary);
        }
        return result;
    }
};

int main() {
    Solution sol;
    vector<string> res = sol.binstr(3);
    for (auto &s : res) cout << s << " ";
    return 0;
}
